Sources/Orbit/Components/InputField.swift

```diff
@@ -37,25 +39,58 @@ public struct InputField: View {
             
             InputContent(
                 prefix: prefix,
-                suffix: suffix,
+                suffix: isSecure ? securedSuffix : suffix,
                 state: state,
                 message: message,
                 isEditing: isEditing,
-                suffixAction: suffixAction
-            ) {
-                HStack(spacing: 0) {
-                    textField
-                    Spacer(minLength: 0)
-                    clearButton
+                suffixAction: {
+                    if isSecure {
+                        secureModel.isShowSecuredText.toggle()
+                    }
+                    suffixAction()
+                },
+                label: {
+                    HStack(spacing: 0) {
+                        if isSecure {
+                            secureField
+                        } else {
+                            textField
+                            Spacer(minLength: 0)
+                            clearButton
+                        }
+                    }
                 }
-            }
-            
+            )
+
             ContentHeightReader(height: $messageHeight.animation(.easeOut(duration: 0.2))) {
                 FormFieldMessage(message)
             }
         }
     }
 
+    @ViewBuilder var secureField: some View {
+        SecureField(
+            "",
+            text: $value
+        ).onTapGesture {
+            self.isEditing = isEditing
+            onEditingChanged(isEditing)
+        }
+        .textFieldStyle(TextFieldStyle(leadingPadding: 0))
+        .autocapitalization(autocapitalization)
+        .disableAutocorrection(isAutocompleteEnabled == false)
```

Comment:
Probably most of these modifiers are the same as for non-secure version. It should be extracted to vary only what is truly different between those two.

Reply:
What is the way to achieve? Should it be something like
```
if isSecured {
   SecureField()
   .customeModifier1
} else {
   TextField
   .customeModifier2
}
.commonModifier1
.commonModifier2
```

Reply:
Yes exactly 👍

If it gets too long, it can be further wrapped as 

```
var body: some View {
   if isSecured {
      secureField
   } else {
      textField
}

var secureField: some View {
   inputField
      .customMod2
}

var textField: some View {
   inputField
      .customMod1
}

var inputField: some View {
   //common code
}
```

Reply:
I've decided to extract field into separate var:
```
    @ViewBuilder var input: some View {
        if isSecure {
            secureField
        } else {
            textField
        }
    }
```

thought can't come up with some meaningful name...

Anyway, compiler doesn't allow me to use ternary operator saying 
> Result values in '? :' expression have mismatching types 'some View' (type of 'InputField.secureField') and 'some View' (type of 'InputField.textField')

for `isSecure ? secureField : textField`


Do you know why?

Reply:
This is a function builder, so the whole thing desugars into method calls. Perhaps ternary operator support is not implemented.